{doc} Add aaz-dev-tools in authoring commands page (#26691)
Azure/azure-cli: doc/authoring_command_modules/authoring_commands.md

The document provides instructions and guidelines on how to author individual commands.

> > **Note**

Reviewer: Is it better to list the benefits here as well as below?
```
[new codegen tool](https://azure.github.io/aaz-dev-tools/) is UI based and easy to use. We encourage your team to use it to generate CLI commands automatically as nobody knows your logic better than your team. Using our new codegen tool, you can have the following benefits:

- Shorten development cycle (save 50%+ development time, if no block issues, usually it takes less than 1 week).
- Directly generate CLI commands from swagger instead of waiting for SDK release.
- Consistent code style improves command quality and avoids code review efforts.
```

> 

> > [aaz-dev-tools](https://azure.github.io/aaz-dev-tools/) is our codegen tool which can be used for Control Plane APIs. Highly recommend using it in new commands.

Author: ```suggestion
> [aaz-dev-tools](https://azure.github.io/aaz-dev-tools/) is our codegen tool which can be used for Control Plane APIs. Highly recommend using it in new commands. It brings the following benefits:
> - Shorten development cycle (save 50%+ development time).
> - Consistent code style improves command quality and saves PR review time.
> - Directly generate CLI commands from swagger instead of waiting for SDK release.
```

> > If you have any issues, you can post it in Teams [channel](https://aka.ms/azure-cli-codegen-channel).

Author: ```suggestion
> If you have any question, you can post it in this Teams [channel](https://aka.ms/azure-cli-codegen-channel).
```

The document provides instructions and guidelines on how to author individual commands.

> **Note**
> 
> [aaz-dev-tools](https://azure.github.io/aaz-dev-tools/) is our codegen tool which can be used for Control Plane APIs. Highly recommend using it in new commands. It brings the following benefits:
> - Shorten development cycle (save 50%+ development time).
> - Consistent code style improves command quality and saves PR review time.

> > - Directly generate CLI commands from swagger instead of waiting for SDK release.

Author: ```suggestion
> - Directly generate CLI commands from swagger instead of waiting for SDK release.
>
```
